File: submissions/reservoir-coders/Challenge6.cs
static void f(int c)
{
    int tip;
    if (c<2000)
    {
        tip = (100 - (c % 100)) % 100;
    }
    else if (c<5000)
    {
        tip = (500 - (c % 500)) % 500;
    }
    else
    {
        tip = (int)(c / 10);
        var a = (c+tip) % 500;
        if (a>250)
        {
            tip += 500-a;
    	}
        else
        {
            tip -= a;
        }
    }
    if (tip == 0 || tip > 1500)
    {
        Console.WriteLine("PAID BY CARD");
    }
    else if (tip % 100 > 0)
    {
        Console.WriteLine("KEEP THE CHANGE YOU FILTY ANIMAL");
    }
    else
    {
        Console.WriteLine("HERE IS {0}", tip+c);
    }
}

File: submissions/bat/challenge6.c
void f(int c){int s=(c<2000)?g(c,100):(c<5000)?g(c,500):r(c*1.1/500.0)*500;int t=s-c;printf((t==0||t>1500)?"PAID BY CARD":(t%100!=0)?"KEEP THE CHANGE YOU FILTY ANIMAL":"HERE IS %d",s);}
int g(int c,int r){return c%r==0?c:(c+r)/r*r;}
int r(double x){if(x<0.0)return(int)(x-0.5);else return(int)(x+0.5);}
